### PR TITLE
[Silabs]Rework EFR32 spi com to rs9116/siwx917 wifi ncp

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
Rework EFR32 platforms + wifi ncp radio rs9116 or siwx917 spi communication to use our `spidrv` and improve the synchronization mechanism for the dma transfer

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by jmartinez-silabs** [comment]
Thanks @bzbarsky-apple , I believe it is a pretty straightforward cleanup and I'll ask for a thorough review from Silabs folks. It has also been tested in our internal wifi testbench Thanks!

File: examples/platform/silabs/efr32/rs911x/hal/efx_spi.c [hunk 1]
```diff
@@ -163,115 +150,24 @@ void rsi_hal_board_init(void)
...
-
-    (void) channel;
-    (void) sequenceNo;
-    (void) userParam;
-
-    // WFX_RSI_LOG ("SPI: DMA done [%x,%x,%x,%x]", buf [0], buf [1], buf [2], buf [3]);
-    xSemaphoreGiveFromISR(spi_sem, &xHigherPriorityTaskWoken);
+    vTaskNotifyGiveFromISR(spiInitiatorTaskHandle, &xHigherPriorityTaskWoken);
+    spiInitiatorTaskHandle = NULL;
     portYIELD_FROM_ISR(xHigherPriorityTaskWoken);
```
Comment (jmeg-sfy) [inline examples/platform/silabs/efr32/rs911x/hal/efx_spi.c:170]: so this is always called from an ISR context ?

File: examples/platform/silabs/efr32/rs911x/hal/efx_spi.c [hunk 2]
```diff
@@ -287,57 +183,60 @@ static void transmitDMA(uint8_t * rx_buf, uint8_t * tx_buf, uint16_t xlen)
...
+    }
+    else if (rx_buf == NULL) // Tx operation only
+    {
+        spiError = SPIDRV_MTransmit(SPI_HANDLE, tx_buf, xlen, spi_dmaTransfertComplete);
+    }
+    else // Tx and Rx operation
+    {
+        spiError = SPIDRV_MTransfer(SPI_HANDLE, tx_buf, rx_buf, xlen, spi_dmaTransfertComplete);
+    }
+
```
Comment (jmeg-sfy) [inline examples/platform/silabs/efr32/rs911x/hal/efx_spi.c:215]: why the SPIDRV doesn't provide an helper for this within the SDK

**Review by jmeg-sfy — APPROVED** [review]

File: examples/platform/silabs/efr32/rs911x/hal/efx_spi.c (hunk 1 above)
Comment (jmartinez-silabs) [reply examples/platform/silabs/efr32/rs911x/hal/efx_spi.c:170]: Yes it is a DMA completion callback the DMA completion is triggered by a IRQ

File: examples/platform/silabs/efr32/rs911x/hal/efx_spi.c (hunk 2 above)
Comment (jmartinez-silabs) [reply examples/platform/silabs/efr32/rs911x/hal/efx_spi.c:215]: you'd have to ask the platform team that implemented the spidrv. I don't have an answer

**Review by mkardous-silabs — APPROVED** [review]

**Pull request merged**